# BASIC GLS PERIODOGRAMS RESULTS ANALYSIS FOR CARMENES TESS CURVES

We review the basic results of the GLS periodograms applied over the TESS curves of CARMENES objects, specially looking at the wrong records.

**Note:** In this notebook only the 269 objects in common with RV are analyzed.

## Modules and configuration

### Modules

In [1]:
import pandas as pd

from scipy.stats import normaltest

import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
plt.figure(figsize = (15, 10), dpi = 80)

import seaborn as sns
sns.set_style("white", {'figure.figsize':(15,10)})

<Figure size 1200x800 with 0 Axes>

### Configuration

In [2]:
GTO_FILE = "../data/CARM_VIS_objects_with_PG.csv"
#GTO_FILE = "../data/GTO_objects_with_PG.csv"
#GTO_FILE = "../data/SELECTION_GTO_objects_with_PG.csv"
RV_FOLDER = "../data/CARMENES_GTO_RVs/"

IMAGE_FOLDER = "./img/"

ALPHA = 1e-3 # Significance threshold for statistical tests

### Functions

In [3]:
def draw_hist(d1, d1_label, fig_filename=None):
    '''Draws an histogram of the values passed. It also calulates the statistics
    and tests the goodness of fit by a Gaussian distribution'''
    # Plots and saves the figure
    kwargs = dict(hist_kws={'alpha':.4}, kde_kws={'linewidth':2})
    plt.figure(figsize=(10,7))
    plt.title(d1_label + " values distribution", fontsize=16)
    plt.grid(axis='both', alpha=0.75)
    plt.xlabel(d1_label, fontsize=12)
    plt.ylabel('Frequency', fontsize=12)
    sns.distplot(d1, label=d1_label, **kwargs)
    #sns.histplot(d1, label=d1_label)
    #plt.xlim(0,100)
    #plt.legend(loc='upper right');
    # Save the image:
    if fig_filename is None:
        pass
    else:
        plt.savefig(fig_filename, format='png')
    # Display basic statistics:
    print(pd.Series(d1).describe())
    # Perform the normaltest (D'Agostino and Pearson) for Gaussian distribution fit:
    k2, p = normaltest(d1)
    alpha = ALPHA
    print("p = {:g}".format(p))
    if p < alpha:  # null hypothesis: x comes from a normal distribution
        print("***This distribution does NOT come from a Gaussian distribution" \
              "(with significance level = %.6f)" %(1.0 - alpha))
    else:
        print("This distribution MIGHT COME come from Gaussian distribution" \
              "(with significance level = %.6f)" %(1.0 - alpha))

## Load CARM\_VIS table with GLS results

In [4]:
gto = pd.read_csv(GTO_FILE, sep=',', decimal='.')
gto.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
0  J23585+076        Wolf 1051   AB  4383    359.636420      7.656947   
1  J23556-061           GJ 912   AB   912    358.913617     -6.144283   
2  J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3  J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
4  J23492+024           BR Psc    -   908    357.306604      2.396918   

      RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ... WF_offset_PG_TESS  \
0  23:58:32.65  +07:39:30.1   100.839686   -52.931049  ...               NaN   
1  23:55:39.78  -06:08:33.4    88.129933   -65.175491  ...               NaN   
2  23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...        999.999756   
3  23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...       1000.000122   
4  23:49:12.53  +02:24:04.4    93.567467   -56.885396  ...               NaN   

  WF_e_offset_PG_TESS  WF_FAP_PG_TESS WF_valid_PG_TESS  \
0                 NaN             NaN              0.0   
1                 NaN             NaN              0.0   
2        2.151008e-06             1.0              1.0   
3        9.022946e-07             1.0              1.0   
4                 NaN             NaN              0.0   

                                    WF_error_PG_TESS  WF_elapsed_time_PG_TESS  \
0  Not recognized as a supported data product:\nn...                 0.000997   
1  Not recognized as a supported data product:\nn...                 0.000998   
2                                                NaN                94.758838   
3                                                NaN               132.607176   
4  Not recognized as a supported data product:\nn...                 0.001995   

                                  WF_plain_file_TESS  \
0                                                NaN   
1                                                NaN   
2  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23548+38...   
3  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23505-09...   
4                                                NaN   

                                    WF_fig_file_TESS  \
0                                                NaN   
1                                                NaN   
2  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...   
3  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...   
4                                                NaN   

                                      PG_file_RV  \
0  ../data/CARM_VIS_RVs_PGs/J23585+076_RV_PG.dat   
1  ../data/CARM_VIS_RVs_PGs/J23556-061_RV_PG.dat   
2  ../data/CARM_VIS_RVs_PGs/J23548+385_RV_PG.dat   
3  ../data/CARM_VIS_RVs_PGs/J23505-095_RV_PG.dat   
4  ../data/CARM_VIS_RVs_PGs/J23492+024_RV_PG.dat   

                                     PG_file_TESS  
0                                             NaN  
1                                             NaN  
2  ../data/CARM_VIS_TESS_PGs/J23548+385_RV_PG.dat  
3  ../data/CARM_VIS_TESS_PGs/J23505-095_RV_PG.dat  
4                                             NaN  

[5 rows x 300 columns]

In [5]:
print(list(gto.columns))

['Karmn', 'Name', 'Comp', 'GJ', 'RA_J2016_deg', 'DE_J2016_deg', 'RA_J2000', 'DE_J2000', 'l_J2016_deg', 'b_J2016_deg', 'Ref01', 'SpT', 'SpTnum', 'Ref02', 'Teff_K', 'eTeff_K', 'logg', 'elogg', '[Fe/H]', 'e[Fe/H]', 'Ref03', 'L_Lsol', 'eL_Lsol', 'Ref04', 'R_Rsol', 'eR_Rsol', 'Ref05', 'M_Msol', 'eM_Msol', 'Ref06', 'muRA_masa-1', 'emuRA_masa-1', 'muDE_masa-1', 'emuDE_masa-1', 'Ref07', 'pi_mas', 'epi_mas', 'Ref08', 'd_pc', 'ed_pc', 'Ref09', 'Vr_kms-1', 'eVr_kms-1', 'Ref10', 'ruwe', 'Ref11', 'U_kms-1', 'eU_kms-1', 'V_kms-1', 'eV_kms-1', 'W_kms-1', 'eW_kms-1', 'Ref12', 'sa_m/s/a', 'esa_m/s/a', 'Ref13', 'SKG', 'Ref14', 'SKG_lit', 'Ref14_lit', 'Pop', 'Ref15', 'vsini_flag', 'vsini_kms-1', 'evsini_kms-1', 'Ref16', 'P_d', 'eP_d', 'Ref17', 'pEWHalpha_A', 'epEWHalpha_A', 'Ref18', 'log(LHalpha/Lbol)', 'elog(LHalpha/Lbol)', 'Ref19', '1RXS', 'CRT_s-1', 'eCRT_s-1', 'HR1', 'eHR1', 'HR2', 'eHR2', 'Flux_X_E-13_ergcm-2s-1', 'eFlux_X_E-13_ergcm-2s-1', 'LX/LJ', 'eLX/LJ', 'Ref20', 'Activity', 'Ref21', 'FUV_mag',

### Addendum: correct some RV column names

In [6]:
if 'f_RV' in list(gto.columns):
    print("Correcting column names...")
    gto.rename(columns={'f_RV': 'f_PG_RV', 'e_f_RV': 'e_f_PG_RV',
                        'Pd_RV': 'Pd_PG_RV', 'e_Pd_RV': 'e_Pd_PG_RV',
                        'Ph_RV': 'Ph_PG_RV', 'e_Ph_RV': 'e_Ph_PG_RV',
                        'Pm_RV': 'Pm_PG_RV', 'e_Pm_RV': 'e_Pm_PG_RV',
                        'A_RV': 'A_PG_RV', 'e_A_RV': 'e_A_PG_RV',
                        'ph_RV': 'ph_PG_RV', 'e_ph_RV': 'e_ph_PG_RV',
                        'T0_RV': 'T0_PG_RV', 'e_T0_RV': 'e_T0_PG_RV',
                        'offset_RV': 'offset_PG_RV', 'e_offset_RV': 'e_offset_PG_RV',
                        'FAP_RV': 'FAP_PG_RV', 'elapsed_time_RV': 'elapsed_time_PG_RV'},
               inplace=True)
    gto.to_csv(GTO_FILE, sep=',', decimal='.', index=False)
else:
    print("No correction needed.")
gto.head()

No correction needed.


Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
0  J23585+076        Wolf 1051   AB  4383    359.636420      7.656947   
1  J23556-061           GJ 912   AB   912    358.913617     -6.144283   
2  J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3  J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
4  J23492+024           BR Psc    -   908    357.306604      2.396918   

      RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ... WF_offset_PG_TESS  \
0  23:58:32.65  +07:39:30.1   100.839686   -52.931049  ...               NaN   
1  23:55:39.78  -06:08:33.4    88.129933   -65.175491  ...               NaN   
2  23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...        999.999756   
3  23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...       1000.000122   
4  23:49:12.53  +02:24:04.4    93.567467   -56.885396  ...               NaN   

  WF_e_offset_PG_TESS  WF_FAP_PG_TESS WF_valid_PG_TESS  \
0                 NaN             NaN              0.0   
1                 NaN             NaN              0.0   
2        2.151008e-06             1.0              1.0   
3        9.022946e-07             1.0              1.0   
4                 NaN             NaN              0.0   

                                    WF_error_PG_TESS  WF_elapsed_time_PG_TESS  \
0  Not recognized as a supported data product:\nn...                 0.000997   
1  Not recognized as a supported data product:\nn...                 0.000998   
2                                                NaN                94.758838   
3                                                NaN               132.607176   
4  Not recognized as a supported data product:\nn...                 0.001995   

                                  WF_plain_file_TESS  \
0                                                NaN   
1                                                NaN   
2  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23548+38...   
3  ../data/CARM_VIS_TESS_WinFunc_PGs/WF_J23505-09...   
4                                                NaN   

                                    WF_fig_file_TESS  \
0                                                NaN   
1                                                NaN   
2  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...   
3  ../data/CARM_VIS_TESS_WinFunc_PGs/figures/WF_J...   
4                                                NaN   

                                      PG_file_RV  \
0  ../data/CARM_VIS_RVs_PGs/J23585+076_RV_PG.dat   
1  ../data/CARM_VIS_RVs_PGs/J23556-061_RV_PG.dat   
2  ../data/CARM_VIS_RVs_PGs/J23548+385_RV_PG.dat   
3  ../data/CARM_VIS_RVs_PGs/J23505-095_RV_PG.dat   
4  ../data/CARM_VIS_RVs_PGs/J23492+024_RV_PG.dat   

                                     PG_file_TESS  
0                                             NaN  
1                                             NaN  
2  ../data/CARM_VIS_TESS_PGs/J23548+385_RV_PG.dat  
3  ../data/CARM_VIS_TESS_PGs/J23505-095_RV_PG.dat  
4                                             NaN  

[5 rows x 300 columns]

## Analysis of results

### Exceptions

#### Objects which calculation failed

We first check how many objects had problems and could not complete the calculation.

In [7]:
gto.loc[gto['valid_PG_TESS'] == 0, ['Karmn', 'error_PG_TESS']].shape

(119, 2)

Hence, periodogram calculation for _TESS_ was unsuccessful for 119 objects.

In [8]:
gto.loc[gto['valid_PG_TESS'] == 0, ['Karmn', 'error_PG_TESS']]

Karmn                                      error_PG_TESS
0     J23585+076  Not recognized as a supported data product:\nn...
1     J23556-061  Not recognized as a supported data product:\nn...
4     J23492+024  Not recognized as a supported data product:\nn...
6     J23419+441  Not recognized as a supported data product:\nn...
8     J23351-023  Not recognized as a supported data product:\nn...
..           ...                                                ...
373   J01056+284  Not recognized as a supported data product:\nn...
384   J00184+440  Not recognized as a supported data product:\nn...
386  J00162+198W  Not recognized as a supported data product:\nn...
387  J00162+198E  Not recognized as a supported data product:\nn...
388   J00067-075  Not recognized as a supported data product:\nn...

[119 rows x 2 columns]

In general, these errors happened because the _TESS_ light curve for the object was not available.

#### Objects that take too long to complete

During calculation, it was also observed that some objects took too long time to complete. It would be advisable to also check those ones.

In [9]:
gto[['Karmn', 'elapsed_time_PG_TESS', 'valid_PG_TESS', 'error_PG_TESS']] \
    .sort_values(by='elapsed_time_PG_TESS', ascending=False)

Karmn  elapsed_time_PG_TESS  valid_PG_TESS  \
297  J06011+595            114.964168            1.0   
295  J06103+821            114.598526            1.0   
298  J06000+027            111.330960            1.0   
292  J06318+414            110.901375            1.0   
296  J06024+498            108.847631            1.0   
..          ...                   ...            ...   
254  J09003+218              0.000963            0.0   
8    J23351-023              0.000963            0.0   
388  J00067-075              0.000961            0.0   
212  J10508+068              0.000957            0.0   
340  J03213+799              0.000954            0.0   

                                         error_PG_TESS  
297                                                NaN  
295                                                NaN  
298                                                NaN  
292                                                NaN  
296                                                NaN  
..                                                 ...  
254  Not recognized as a supported data product:\nn...  
8    Not recognized as a supported data product:\nn...  
388  Not recognized as a supported data product:\nn...  
212  Not recognized as a supported data product:\nn...  
340  Not recognized as a supported data product:\nn...  

[390 rows x 4 columns]

We can see that in this case the large amount of time spent in calculations does not seem to be a problem.

#### Objects which peak frequency is in the lower limit

In [10]:
gto[['Karmn', 'f_PG_TESS', 'e_f_PG_TESS']].dropna().sort_values(by='f_PG_TESS', ascending=False)

Karmn  f_PG_TESS  e_f_PG_TESS
342   J03142+286    287.916     0.004409
202  J11110+304W    287.862     0.000882
303   J05366+112    287.820     0.001984
150   J14251+518    287.790     0.001782
344   J03090+100    287.760     0.005395
..           ...        ...          ...
357   J02336+249      2.448     0.004717
207   J11033+359      2.445     0.004657
354   J02442+255      2.436     0.005168
346  J02565+554W      2.427     0.003011
19    J22468+443      2.421     0.004997

[268 rows x 3 columns]

In [11]:
gto[['Karmn', 'f_PG_TESS', 'e_f_PG_TESS']].dropna().sort_values(by='f_PG_TESS', ascending=True)

Karmn  f_PG_TESS  e_f_PG_TESS
19    J22468+443      2.421     0.004997
346  J02565+554W      2.427     0.003011
354   J02442+255      2.436     0.005168
207   J11033+359      2.445     0.004657
357   J02336+249      2.448     0.004717
..           ...        ...          ...
344   J03090+100    287.760     0.005395
150   J14251+518    287.790     0.001782
303   J05366+112    287.820     0.001984
202  J11110+304W    287.862     0.000882
342   J03142+286    287.916     0.004409

[268 rows x 3 columns]

**OBSERVATION:** As we can see, there seem to be at least ten stars (probable many more) for which the calculated pulsation frequencies lie very near the limits of the periodogram range of $P\in[2.4, 288.0]\;d^{-1}$.

In [12]:
gto.loc[(gto['f_PG_TESS'] < 8.0) | (gto['f_PG_TESS'] > 72.0), ['Karmn', 'f_PG_TESS', 'e_f_PG_TESS']]

Karmn  f_PG_TESS  e_f_PG_TESS
2    J23548+385      3.798     0.006598
3    J23505-095      2.793     0.005566
5    J23431+365      4.290     0.004984
7    J23381-162     78.036     0.005545
10   J23245+578      3.183     0.005678
..          ...        ...          ...
380  J00570+450      2.994     0.003871
381  J00403+612    145.164     0.006423
382  J00389+306      3.171     0.005620
383  J00286-066      5.997     0.006868
385  J00183+440    262.377     0.006433

[236 rows x 3 columns]

**OBSERVATION:** there seems to be a lot of stars (236 out of 268; $\approx88\%$) for which the calculated pulsation frequency  lies outside the theoretical predicted pulsation range ($P_{pulsation}\in[8.0, 72.0]\;d^{-1}$).

### Main statistics

In [13]:
gto[['n_TESS', 'Ps_TESS', 'wmean_TESS', 'wrms_TESS', 'maxSNR_PG_TESS', 'Pm_PG_TESS', 'FAP_PG_TESS']].describe()

n_TESS     Ps_TESS     wmean_TESS    wrms_TESS  maxSNR_PG_TESS  \
count    271.000000  271.000000     271.000000   271.000000      270.000000   
mean   14242.291513    0.003121   53071.216947   129.125485       44.441794   
std     4637.616295    0.005445   98538.640302   574.939825      317.527464   
min      694.000000    0.001389 -315455.875000     2.188334        0.014799   
25%    13420.000000    0.001389    8366.517090    16.475475        0.050413   
50%    15602.000000    0.001389   20685.371094    29.356222        0.057367   
75%    16941.500000    0.001389   56393.232422    68.246506        0.099981   
max    18711.000000    0.020834  661204.500000  8249.875000     3969.927358   

       Pm_PG_TESS   FAP_PG_TESS  
count  271.000000  2.700000e+02  
mean   263.327033  1.888368e-01  
std    220.784475  2.901528e-01  
min      5.000000  0.000000e+00  
25%     13.899229  7.175419e-10  
50%    290.381125  3.225141e-02  
75%    471.513225  2.574146e-01  
max    600.000000  1.000000e+00

In [14]:
gto[['n_TESS', 'Ps_TESS']].describe()

n_TESS     Ps_TESS
count    271.000000  271.000000
mean   14242.291513    0.003121
std     4637.616295    0.005445
min      694.000000    0.001389
25%    13420.000000    0.001389
50%    15602.000000    0.001389
75%    16941.500000    0.001389
max    18711.000000    0.020834

In [15]:
print(gto[['n_TESS', 'Ps_TESS']].describe().to_latex(float_format='%.6f'))

\begin{tabular}{lrr}
\toprule
{} &       n\_TESS &    Ps\_TESS \\
\midrule
count &   271.000000 & 271.000000 \\
mean  & 14242.291513 &   0.003121 \\
std   &  4637.616295 &   0.005445 \\
min   &   694.000000 &   0.001389 \\
25\%   & 13420.000000 &   0.001389 \\
50\%   & 15602.000000 &   0.001389 \\
75\%   & 16941.500000 &   0.001389 \\
max   & 18711.000000 &   0.020834 \\
\bottomrule
\end{tabular}



**IMPORTANT NOTE:** the remaining sections were not important for the study, as its main focus lies in the CARM\_VIS RV curves, not in TESS light curves, so the rest of the notebook sections were not coded. The code would be similar to that used for the CARM\_VIS data.